activity/coordinated_worker.go

```diff
@@ -17,101 +17,105 @@ const (
 
 func (w *ActivityWorker) AddCoordinatedHandler(heartbeatInterval time.Duration, handler *CoordinatedActivityHandler) {
 
-	ticks := make(chan CoordinatedActivityHandlerTick)
-	startErr := make(chan error)
+	handlerFunc := func(ctx *ActivityContext, input interface{}) {
+		ticks := make(chan CoordinatedActivityHandlerTick)
+		startErr := make(chan error)
 
-	handlerTickFunc := func(activityTask *swf.ActivityTask, input interface{}) {
-		go func() {
-			cont, res, err := handler.Tick(activityTask, input)
-			ticks <- CoordinatedActivityHandlerTick{
-				Continue: cont,
-				Result:   res,
-				Error:    err,
-			}
-		}()
-	}
+		handlerTickFunc := func(cxt *ActivityContext, input interface{}) {
+			go func() {
+				cont, res, err := handler.Tick(cxt, input)
+				ticks <- CoordinatedActivityHandlerTick{
```

Comment:
since `ticks` is non-buffered, you may want to not block here. Ticks are consumed on three different places in the main goroutine, but the main goroutine can block on some places (such as when calling the SWF API), which would prevent `handler.Tick()` to be called while the main worker goroutine is blocked.

Is this intended (i.e.: better to block ticking until all tick results have been successfully processed), or would it be better to keep calling `handler.Tick` to let handlers keep doing periodic work? (and add a buffer to the channel, or drop ticks that can't be processed and would cause writing to the `ticks` channel to block?